### DISCUSSION TITLE
Provide Actions in template

### DISCUSSION THREAD

**Discussion by mihai-sysbio** [MetabolicAtlas/standard-GEM#30] [Ideas]
Should `standard-GEM` provide a default GitHub Actions that implement `yaml` format validation and/or look at the `cobrapy` import?

**Comment by mihai-sysbio** [comment]
My focus is on making `standard-GEM` accessible, so my fear is that packaging Actions in the template introduces complexity. It would expect users to understand how the Actions work, and also to maintain them if (when) they break.
Instead, I would be very much in favor of setting up a centralized system for validating `standard-GEM`-labelled repositories, even if the feedback loop would be less tight.

**Reply by haowang-bioinfo** [reply]
Would suggest to begin the implementation of `yaml` validation via `cobrapy` import

**Reply by mihai-sysbio** [reply]
There have been cases where these Actions broke. How can we avoid pushing this maintenance responsibility to the adopters of `standard-GEM`?

**Reply by haowang-bioinfo** [reply]
> There have been cases where these Actions broke.

For example?

**Reply by mihai-sysbio** [reply]
Off the top of my head, there's [this PR in Human-GEM](https://github.com/SysBioChalmers/Human-GEM/pull/211).

**Reply by haowang-bioinfo** [reply]
Then one should calculate the cost that will be required to maintain such `a centralized system`, in comparison to the effort in dealing with cases like `swiglpk dependency `

**Reply by mihai-sysbio** [reply]
The centralized system is essentially a for loop after which the "normal" Actions would be applied to each model, so the maintenance cost shouldn't be that high.
Another advantage is that centrally far more checks can be applied than one would be willing to maintain in a per-repo setup.
Last but not least, I think a version of this centralized system is needed to ensure adherence to the `standard-GEM` spec.

**Reply by haowang-bioinfo** [reply]
> The centralized system is essentially a for loop after which the "normal" Actions would be applied to each model, so the maintenance cost shouldn't be that high.

look forward to such a lightweight centralized system

**Reply by mihai-sysbio** [reply]
The work had been started in a branch of this repository. For clearer separation and better visibility, it has been migrated to a new repository [standard-GEM-validation](https://github.com/MetabolicAtlas/standard-GEM-validation).
